default.nix

```diff
@@ -1,11 +1,11 @@
 { system ? builtins.currentSystem , ... }:
 
 let
-  rev = "5272327b81ed355bbed5659b8d303cf2979b6953";
+  rev = "8e78c2cfbae71720869c35b6710d652bf28b37cc";
   nixpkgs = builtins.fetchTarball {
-    name = "nixpkgs-release-20.03";
+    name = "nixpkgs-release-20.09";
     url = "https://github.com/nixos/nixpkgs/tarball/${rev}";
-    sha256 = "0182ys095dfx02vl2a20j1hz92dx3mfgz2a6fhn31bqlp1wa8hlq";
+    sha256 = "178dr6bz71lbv0ljynvvkrc2p0lwqmci482brkqdw9qfx3sc1a7f";
```

Comment:
I think this bump allows us to drop some of the custom packages we've been keeping. `z3` and `cvc4` are up to par or newer on `20.09` so we can drop our own versions. Looking at our other custom packages, we may even be able to drop `libff`, but that might be more housecleaning than what we wanna do in this PR

Reply:
yeah lets tackle that in a seperate PR.

Reply:
lets still drop z3 and cvc4 here tho